qaequilibrae/modules/processing_provider/RenumberFromCentroids.py

```diff
@@ -0,0 +1,129 @@
+__author__ = 'Arthur Evrard'
+
+import importlib.util as iutil
+import pandas as pd
+import sys
+from os.path import join
+from qgis.core import QgsCoordinateReferenceSystem, QgsCoordinateTransform, QgsProject
+from qgis.core import QgsProcessing, QgsProcessingMultiStepFeedback
+from qgis.core import QgsProcessingParameterVectorLayer, QgsProcessingParameterField, QgsProcessingParameterFile, \
+    QgsProcessingParameterString
+from shapely.wkt import loads as load_wkt
+
+from .translatableAlgo import TranslatableAlgorithm
+
+
+class RenumberFromCentroids(TranslatableAlgorithm):
+
+    def initAlgorithm(self, config=None):
+        self.addParameter(QgsProcessingParameterVectorLayer('nodes', self.tr('Centroids'), types=[QgsProcessing.TypeVectorPoint], defaultValue='nodes'))
+        self.addParameter(QgsProcessingParameterField('node_id', self.tr('Zone id'), type=QgsProcessingParameterField.Numeric, parentLayerParameterName='nodes', allowMultiple=False, defaultValue='node_id'))
+        self.addParameter(QgsProcessingParameterFile('PrjtPath', self.tr('AequilibraE project'), behavior=QgsProcessingParameterFile.Folder, defaultValue='D:/'))
+
+    def processAlgorithm(self, parameters, context, model_feedback):
+        feedback = QgsProcessingMultiStepFeedback(3, model_feedback)
+        results = {}
+        outputs = {}
+        
+        # Checks if we have access to aequilibrae library
+        has_aeq = iutil.find_spec("aequilibrae") is not None
+        if not has_aeq:
+            sys.exit(tr('AequilibraE library not found'))
+
+        from aequilibrae import Project
+        feedback.pushInfo(tr('Connecting to aequilibrae project'))
+        project = Project()
+        project.open(parameters['PrjtPath'])
+        feedback.pushInfo(' ')
+        feedback.setCurrentStep(1)
+        
+        feedback.pushInfo(tr('Importing node layer from QGIS'))
+        layer_crs = self.parameterAsVectorLayer(parameters, 'nodes', context).crs()
+        aeq_crs = QgsCoordinateReferenceSystem("EPSG:4326")
+        
+        #Import QGIS layer as a panda dataframe
+        layer = self.parameterAsVectorLayer(parameters, 'nodes', context)
+        columns = [f.name() for f in layer.fields()] + ['geometry']
+        columns_types = [f.typeName() for f in layer.fields()] #
+        row_list = []
+        for f in layer.getFeatures():
+            geom=f.geometry()
+            geom.transform(QgsCoordinateTransform(layer_crs, aeq_crs, QgsProject.instance()))
+            row_list.append(dict(zip(columns, f.attributes() + [geom.asWkt()])))
```

Comment:
I'll often just use a list comp + helper function for things like this

```suggestion
        def fn(f):
            geom=f.geometry()
            geom.transform(QgsCoordinateTransform(layer_crs, aeq_crs, QgsProject.instance()))
            return dict(zip(columns, f.attributes() + [geom.asWkt()]))
        row_list = [fn(f) for f in layer.getFeatures()]
```